mathjax3-ts/a11y/explorer/Region.ts

```diff
@@ -335,36 +334,67 @@ export class HoverRegion extends AbstractRegion {
    */
   constructor(public document: A11yDocument) {
     super(document);
-    this.div.style.fontSize = '800%';
+    this.div.style.fontSize = document.options.a11y.magnify + '%';
     this.inner.classList.add('MJX-TEX');
   }
 
   /**
-   * Shows the live region as a subtitle of a node.
-   * @override
+   * Sets the position of the region with respect to align parameter.  There are
+   * three options: top, bottom and center. Center is the default.
+   *
+   * @param {HTMLElement} node The node that is displayed.
    */
-  public Show(node: HTMLElement, highlighter: sre.Highlighter) {
-    super.Show(node, highlighter);
-  }
-
-
   protected position(node: HTMLElement) {
-    const rect = node.getBoundingClientRect();
-    const top = rect.top;
-    const left = rect.left;
+    const nodeRect = node.getBoundingClientRect();
+    const divRect = this.div.getBoundingClientRect();
+    const xCenter = nodeRect.left + (nodeRect.width / 2);
+    let left = xCenter - (divRect.width / 2);
+    left = (left < 0) ? 0 : left;
+    left = left + window.pageXOffset;
+    let top;
+    switch (this.document.options.a11y.align) {
+    case 'top':
+      top = nodeRect.top - divRect.height - 10 ;
+      break;
+    case 'bottom':
+      top = nodeRect.bottom + 10;
+      break;
+    case 'center':
+    default:
+      const yCenter = nodeRect.top + (nodeRect.height / 2);
+      top = yCenter - (divRect.height / 2);
+    }
+    top = top + window.pageYOffset;
+    top = (top < 0) ? 0 : top;
     this.div.style.top = top + 'px';
     this.div.style.left = left + 'px';
   }
 
+  /**
+   * @override
+   */
   protected highlight(highlighter: sre.Highlighter) {
-    // const color = highlighter.colorString();
-    // this.inner.style.backgroundColor = color.background;
-    // this.inner.style.color = color.foreground;
+    const color = highlighter.colorString();
+    this.inner.style.backgroundColor = color.background;
+    this.inner.style.color = color.foreground;
   }
 
-  public AddNode(node: HTMLElement): void {
+  /**
+   * @override
+   */
+  public Show(node: HTMLElement, highlighter: sre.Highlighter) {
+    this.AddNode(node);
+    super.Show(node, highlighter);
+  }
+
+  /**
+   * Adds a clone of the given node to the hover region.
+   * @param {HTMLElement} node The current HTML node.
+   */
+  public AddNode(node: HTMLElement) {
     this.Clear();
-    this.inner.appendChild(node);
+    let mjx = node.cloneNode(true) as HTMLElement;
+    this.inner.appendChild(mjx);
```

Comment:
Unfortunately, simply cloning the element without the proper containers will not get the spacing and sizing correct.  A number of the CSS rules rely on the `mjx-container` and also on the `mjx-math` element being a parent of the element being displayed.  So you will probably need to include these (when they aren't already present).  These sometimes have attributes that you want to include as well (like font size, and display settings), so you probably need to find the actual elements and clone them (shallowly).

Here is an example that does it, though it is not pretty:

```
    public AddNode(node: HTMLElement) {
        this.Clear();
        let mjx = node.cloneNode(true) as HTMLElement;
        if (mjx.nodeName !== 'MJX-CONTAINER') {
            mjx.style.marginLeft = max-style.marginRight = '0';                // remove element spacing (could be done in CSS)
            let container = node;
            while (container && container.nodeName !== 'MJX-CONTAINER') {
                container = container.parentNode as HTMLElement;
            }
            if (mjx.nodeName !== 'MJX-MATH') {
                mjx = container.firstChild.cloneNode(false).appendChild(mjx).parentNode as HTMLElement;
            }
            mjx = container.cloneNode(false).appendChild(mjx).parentNode as HTMLElement;
            mjx.style.margin = '0';                 //  remove displayed math margins (could be done in CSS)
        }
        this.inner.appendChild(mjx);
        this.inner.style.lineHeight = '0';               // could be moved to when this.inner is created
    }
```

I've also removed the spacing that is added between elements, and above and below displayed expressions.  Setting the `line-height` to 0 avoids unwanted extra space below the expression.  Finally, you can remove the `MJX-TEX` class from the class list at line 368, as that is handled by the `mjx-math` element.

Reply:
I've integrated your suggestions and added some support for SVG. The SVG area is still too large, but at least displays the correct content. I am not going to spend any more time on  that right now, though, and leave it as a future issue.

Reply:
I've made a PR to this branch that handles the SVG bounding box, and noted some other problems.  The main one is that tagged equations in SVG don't get processed properly, and that might want to be addressed for v3.  The other issues can wait.